docs/ref/models/fields.txt

```diff
@@ -1031,6 +1031,59 @@ are converted to lowercase.
 If you allow for blank values, you have to allow for null values since blank
 values are stored as null.
 
+``JSONField``
+-------------
+
+.. class:: JSONField(encoder=None, decoder=None, **options)
+
+A field for storing JSON encoded data. In Python the data is represented in
+its Python native format: dictionaries, lists, strings, numbers, booleans
+and ``None``.
+
+.. attribute:: JSONField.encoder
+
+    An optional JSON-encoding class to serialize data types not supported
+    by the standard JSON serializer (``datetime``, ``uuid``, etc.). For
+    example, you can use the
+    :class:`~django.core.serializers.json.DjangoJSONEncoder` class.
+
+    If ``None`` is specified, it defaults to :py:class:`json.JSONEncoder`.
+
+.. attribute:: JSONField.decoder
+
+    An optional JSON-decoding class to deserialize the value retrieved from
+    the database. The value will be in the format chosen by the custom encoder
+    (most often a string). Your deserialization may need to account for the
+    fact that you can't be certain of the input type. For example, you run the
+    risk of returning a ``datetime`` that was actually a string that just
+    happened to be in the same format chosen for ``datetime``\s.
+
+    If ``None`` is specified, it defaults to :py:class:`json.JSONDecoder`.
```

Comment:
I think this and the above description of `encoder` could be clearer that it should be a subclass of `json.JSONEncoder` / `Decoder`, and that the specification can be found in the Python docs.

Reply:
Thanks, done. Not sure if the "Specification for ``JSONEncoder`` and ``JSONDecoder`` subclasses can be found in the Python documentation." line is necessary since `json.JSONEncoder` and `json.JSONEncoder` already link to Python docs.

Reply:
I think it's useful as otherwise it's not clear where to begin implementing your own.